### PR TITLE
General refactoring

### LINKED ISSUE #54: Repository update improvements
- `add_target_repos` should be able to traverse to `repositories.json` entries and add targets based on that
- rename `add_targets` to `sign_targets`
- rename `add_target_repos` to something like `update_targets`
- Check if everything works when when loading snapshot and timestamp keys from the file system and using YubiKey for signing

### LINKED ISSUE #59: Consistently use pathlib instead of the os module
The majority of `developer_tool` relies on `pathlib`. The exceptions are `build_repository` and `create_repository` functions. Additionally, `os.makedirs` can be replaced with Path's `mkdir`.  We can also use glob with `**`, which should return all this directory and all subdirectories recursively. Since we are transitioning to glob, it might be better to use in places where we later check if a path is a directory. We should also update other modules, like `repository_tool`, and the updater.

### PR DESCRIPTION
**Pull request by danixeee**
## Description (e.g. "Related to ...", etc.)

Closes #54.
Closes #59.

+ some more refactoring.

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: taf/cli.py [hunk 1]
```diff
@@ -42,8 +42,16 @@ def add_target_file(repo_path, file_path, keystore, keys_description, scheme):
...
 @click.option('--namespace', default=None, help='Namespace of the target repositories')
-def add_target_repos(repo_path, targets_dir, namespace):
-    developer_tool.add_target_repos(repo_path, targets_dir, namespace)
+def update_repos_from_fs(repo_path, targets_dir, namespace):
+    "Updates target repositories by traversing given targets directory "
+    developer_tool.update_target_repos_from_fs(repo_path, targets_dir, namespace)
+
+
+@cli.command()
+@click.option("--repo-path", default="repository", help="Location of the repository")
```
Comment (renatav) [inline taf/cli.py]: Missing `targets_dir` and `namespace` click options

File: taf/developer_tool.py [hunk 2]
```diff
@@ -134,27 +154,107 @@ def build_auth_repo(
...
+    # if it does not, generate the keys and print the output
+    else:
+        key = generate_rsa_key()
+        print(f"{role_name} key:\n\n{key['keyval']['private']}\n\n")
+        public_key = private_key = key
+    role_obj.add_verification_key(public_key)
+    role_obj.load_signing_key(private_key)
+
+
 def create_repository(
```
Comment (renatav) [inline taf/developer_tool.py]: Definitely nothing crucial, but we there is one `os.path.isdir` left in this function

File: taf/developer_tool.py [hunk 3]
```diff
@@ -134,27 +154,107 @@ def build_auth_repo(
...
+
+    # set Yubikey expiration date
+    role_obj.add_verification_key(key, expires=YUBIKEY_EXPIRATION_DATE)
+    role_obj.add_external_signature_provider(
+        key, partial(signature_provider, key["keyid"], cert_cn)
+    )
+    yubikeys[role_name][serial_num] = (key, cert_cn)
+
+
+def _register_key(keystore, passwords, role_obj, role_name, key_name, key_num, scheme):
```
Comment (renatav) [inline taf/developer_tool.py]: We could also replace these few `os.path` calls here.

File: taf/developer_tool.py [hunk 4]
```diff
@@ -281,22 +302,22 @@ def create_repository(
 
 def generate_keys(keystore, roles_key_infos):
```
Comment (renatav) [inline taf/developer_tool.py]: One `os.path` call left here

File: .flake8 [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [flake8]
 ignore = E203, E266, E501, W503
 max-line-length = 88
-max-complexity = 20
+max-complexity = 18
```
Comment (renatav) [inline .flake8]: Did you mean to commit this? Is is better t reduce complexity?

Comment (danixeee) [reply .flake8]: That's the number of nested statements, lower is better for code readability.

**Pull request merged**